libraries/AP_HAL_ChibiOS/hwdef/CubeOrange-Ottano/scripts/ICE-auto-warmup.lua

```diff
@@ -0,0 +1,153 @@
+   return Parameter(PARAM_TABLE_PREFIX .. name)
+end
+
+--Setup Warmup Parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 13), 'could not add param table')
+
+--[[
+  // @Param: WARMUP_ENABLED
+  // @DisplayName: Warmup Enabled
+  // @Description: Enables the automatic warmup sequence
+  // @Range: 0 1
+  // @Units: degC
+--]]
+local WARMUP_ENABLED  = bind_add_param('ENABLED', 1, 1)
+--[[
+  // @Param: WARMUP_STEP
+  // @DisplayName: Warmup Step
+  // @Description: Sets the temperature at which the engine will increase RPM to WARMUP_RPM
+  // @Range: 25 100
+  // @Units: degC
+--]]
+local WARMUP_STEP = bind_add_param('STEP', 2, 60)
+--[[
+  // @Param: WARMUP_DONE
+  // @DisplayName: WARMUP_DONE
+  // @Description: Temperature at which the warmup sequence is done
+  // @Range: 100 130
+  // @Units: degC
+--]]
+local WARMUP_DONE  = bind_add_param('DONE', 3, 120)
+--[[
+  // @Param: WARMUP_RPM
+  // @DisplayName: WARMUP_RPM
+  // @Description: RPM to increase to when WARMUP_STEP temperature is reached
+  // @Range: 2400 4000
+  // @Units: degC
+--]]
+local WARMUP_RPM  = bind_add_param('RPM', 4, 3000)
+
+local ICE_IDLE_RPM = bind_param("ICE_IDLE_RPM")
+local idle = ICE_IDLE_RPM:get()
+
+
+local function Kelvin_to_C (temp)
+    return (temp - 273.15)
+end
+
+local function Warmup(min_cht, Warmup_Status)
+    local step_rpm = WARMUP_RPM:get()
+    local first_step = WARMUP_STEP:get()
+    local final_temp = WARMUP_DONE:get()
+
+    if ICE_IDLE_RPM:get() == nil or step_rpm == nil then
+        gcs:send_text(MAV_SEVERITY_INFO, "ICE Warmup: Check Idle governor params")
+        return STATUS_NOT_DONE
+    end
+
+    if min_cht < first_step and Warmup_Status <= STATUS_IGNITION then
+        gcs:send_text(MAV_SEVERITY_INFO, "ICE Warmup: Idle RPM")
+        ICE_IDLE_RPM:set(idle)
+        return STATUS_IDLE
+    elseif min_cht >= first_step and min_cht < final_temp and Warmup_Status <= STATUS_IDLE then
+        gcs:send_text(MAV_SEVERITY_INFO, string.format("ICE Warmup: %d RPM", step_rpm))
+        ICE_IDLE_RPM:set(step_rpm)
+        return STATUS_STEP
+    elseif min_cht >= final_temp and Warmup_Status <= STATUS_STEP then
+        gcs:send_text(MAV_SEVERITY_INFO, "ICE Warmup: Done")
+        ICE_IDLE_RPM:set(idle)
+        return STATUS_DONE
+    end
+    return Warmup_Status
+
+end
+
+-- main update function
+local function update()
+    local engine = efi:get_state()
+    local cylinder_status = engine:cylinder_status()
+    local cht1 = Kelvin_to_C(cylinder_status:cylinder_head_temperature())
+    local cht2 = Kelvin_to_C(cylinder_status:cylinder_head_temperature2())
+    local min_cht = math.min(cht1, cht2)
+    local rpm = engine:engine_speed_rpm()
+
+    --Only run if warmup is enabled and Disarmed
+    if WARMUP_ENABLED:get() == 0 or arming:is_armed() then
+        return
+    end
+
+    if cht1 == nil or cht2 == nil then
```

Comment:
As per docs.lua:
`-- get field
---@return number
function Cylinder_Status_ud:cylinder_head_temperature() end`

This check can be removed. As per docs.lua - cht1, cht2 can never be nil. 
same with rpm and cylinder_status